test/extensions/stats_sinks/hystrix/hystrix_test.cc

```diff
@@ -0,0 +1,433 @@
+    counter.name_ = counter_name;
+    ON_CALL(cluster_stats_scope_, counter(counter_name)).WillByDefault(ReturnRef(counter));
+  }
+
+  void setCountersToZero() {
+    ON_CALL(error_5xx_counter_, value()).WillByDefault(Return(0));
+    ON_CALL(retry_5xx_counter_, value()).WillByDefault(Return(0));
+    ON_CALL(error_4xx_counter_, value()).WillByDefault(Return(0));
+    ON_CALL(retry_4xx_counter_, value()).WillByDefault(Return(0));
+    ON_CALL(success_counter_, value()).WillByDefault(Return(0));
+  }
+
+  // Set counter return values to simulate traffic
+  void setCounterReturnValues(const uint64_t i, const uint64_t success_step,
+                              const uint64_t error_4xx_step, const uint64_t error_4xx_retry_step,
+                              const uint64_t error_5xx_step, const uint64_t error_5xx_retry_step,
+                              const uint64_t timeout_step, const uint64_t timeout_retry_step,
+                              const uint64_t rejected_step) {
+    ON_CALL(error_5xx_counter_, value()).WillByDefault(Return((i + 1) * error_5xx_step));
+    ON_CALL(retry_5xx_counter_, value()).WillByDefault(Return((i + 1) * error_5xx_retry_step));
+    ON_CALL(error_4xx_counter_, value()).WillByDefault(Return((i + 1) * error_4xx_step));
+    ON_CALL(retry_4xx_counter_, value()).WillByDefault(Return((i + 1) * error_4xx_retry_step));
+    ON_CALL(success_counter_, value()).WillByDefault(Return((i + 1) * success_step));
+    cluster_info_->stats().upstream_rq_timeout_.add(timeout_step);
+    cluster_info_->stats().upstream_rq_per_try_timeout_.add(timeout_retry_step);
+    cluster_info_->stats().upstream_rq_pending_overflow_.add(rejected_step);
+  }
+
+  NiceMock<Upstream::MockCluster> cluster_;
+  Upstream::MockClusterInfo* cluster_info_ = new NiceMock<Upstream::MockClusterInfo>();
+  Upstream::ClusterInfoConstSharedPtr cluster_info_ptr_{cluster_info_};
+
+  NiceMock<Stats::MockStore> stats_store_;
+  NiceMock<Stats::MockStore> cluster_stats_scope_;
+  const std::string cluster_name_;
+
+  NiceMock<Stats::MockGauge> membership_total_gauge_;
+  NiceMock<Stats::MockCounter> success_counter_;
+  NiceMock<Stats::MockCounter> error_5xx_counter_;
+  NiceMock<Stats::MockCounter> retry_5xx_counter_;
+  NiceMock<Stats::MockCounter> error_4xx_counter_;
+  NiceMock<Stats::MockCounter> retry_4xx_counter_;
+};
+
+class HystrixSinkTest : public testing::Test {
+public:
+  HystrixSinkTest() { sink_.reset(new HystrixSink(server_, window_size_)); }
+
+  // Return the value corresponding to key in an event stream.
+  absl::string_view getStreamField(absl::string_view data_message, absl::string_view key) {
+    absl::string_view::size_type key_pos = data_message.find(key);
+    EXPECT_NE(absl::string_view::npos, key_pos);
+    absl::string_view trim_data_before_Key = data_message.substr(key_pos);
+    key_pos = trim_data_before_Key.find(" ");
+    EXPECT_NE(absl::string_view::npos, key_pos);
```

Comment:
Probably for all these EXPECTs it would be good to add
```
<< "data_message=" << data_message;
```
In case someone besides you runs into this, that would let them know why the data didn't look like what was expected.